sonar-python-plugin/src/main/java/org/sonar/plugins/python/IPythonNotebookParser.java

```diff
@@ -78,4 +82,23 @@ private static void processCodeCell(JsonParser parser) throws IOException {
       }
     }
   }
+   private static Map<Integer, Integer> countEscapeCharacters(String sourceLine, Map<Integer, Integer> colMap, int colOffSet) {
+    int count = 0;
+    var arr = sourceLine.toCharArray();
+    for (int i = 1; i < sourceLine.length() - 1; ++i) {
+      char c = arr[i];
+      switch (c) {
+        case '"','\'', '/': 
+          // + 1 as we do have to count the open quote.
+          colMap.put(i ,i + colOffSet + count + 1);
+          break;
+        case '\\', '\b', '\f', '\n', '\r', '\t':
```

Comment:
Do we actually need to increase the count for `\n` when integrating the changes from Maksim?
I guess we'd make that into 2 lines of Python code, with their own column offset?

Also, I guess this category concerns characters that are not escaped when converting the JSON string to Java, but are when converted to Python, right? 

We'd probably need to document explicitly all characters that may be escaped in Python and JSON strings, so that we're sure we're not missing one.

Reply:
You are correct the `\n` and `\r` should not update the counter. And it is correct these are characters that are escaped in the Python code